test/test_linting.py

```diff
@@ -678,6 +678,20 @@ def test_lint_pin():
                 - zlib {{CONDA_ZLIB}}*
         ''',
         '''
+        a:
+          meta.yaml: |
+            package:
+              name: a
+              version: '0.1'
+
+            requirements:
+              build:
+                - zlib
+   
+              run:
```

Comment:
To avoid that someone who does some "beautifications" accidentally removes the purposely included empty line, maybe add a comment in the following line, e.g.
```diff
+              run:  # ^-- DO NOT remove the preceding line; it's part of the test case!
```